github-runner-provisioner/github.go

```diff
@@ -3,20 +3,65 @@ package main
 import (
 	"context"
 	"github.com/google/go-github/v48/github"
+	"golang.org/x/exp/slices"
 	"golang.org/x/oauth2"
 )
 
-func getGitHubRunnerToken(ctx context.Context, owner string, repo string) (token string, err error) {
+func getGitHubAPIClient(ctx context.Context) *github.Client {
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: config.GithubToken},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-
 	client := github.NewClient(tc)
+	return client
+}
+
+func getGitHubRunnerToken(ctx context.Context, owner string, repo string) (token string, err error) {
+	client := getGitHubAPIClient(ctx)
 	registrationToken, _, err := client.Actions.CreateRegistrationToken(ctx, owner, repo)
 	if err != nil {
 		return "", err
 	}
 
 	return *registrationToken.Token, nil
 }
+
+func getGitHubRunners(ctx context.Context, owner string, repo string) *github.Runners {
+	client := getGitHubAPIClient(ctx)
+	opts := &github.ListOptions{}
+	runners, _, err := client.Actions.ListRunners(ctx, owner, repo, opts)
```

Comment:
It's common in Go to return an error in addition to the object. This way if there is a problem, the caller doesn't have to check if the object is `nil`, and instead it checks the error.

Also, having an error can provide additional information about the failure, like a message.

Also, having some logs may help identify and troubleshoot issues.